RUMM-581 Add doc file for each extension package (#315)
DataDog/dd-sdk-android: docs/datadog_ndk_module.md

# Datadog NDK extension package

> Provides useful features for applications which include C/C++ code in their code base and not only (e.g. the NDKCrashReporterPlugin could be used to reports native level crashes for any application).

Reviewer: and not only ?

Author: Yeah...it is too familiar I will rephrase it.

> # Datadog NDK extension package

Reviewer 2: ```# Android Native Crash Collection```

Also, instead of `datadog_ndk_module.md` rename this file `native_crash_collection.md`

> <div class="alert alert-info">The Android NDK module is public alpha mode and currently not supported by Datadog.</div>

Reviewer 3: ```suggestion
<div class="alert alert-info">The Android NDK module is in public alpha and not supported by Datadog.</div>
```

> Provides useful features for applications having C/C++ code in their code base but also for the pure Java/Kotlin apps. For example, the NDKCrashReporterPlugin could be used to report native level crashes for any application.

Reviewer 2: > Send crash report for issues rising from the C/C++ code in your application.

(I'm trying here to stay consistent with the tone used in the existing docs)

> Provides useful features for applications having C/C++ code in their code base but also for the pure Java/Kotlin apps (e.g. the NDKCrashReporterPlugin could be used to reports native level crashes for any application).

Reviewer 3: ```suggestion
Provides useful features for applications having C/C++ code in their code base but also for the pure Java/Kotlin apps. For example, the NDKCrashReporterPlugin could be used to report native level crashes for any application.
```

> **Note**: This package is just an extension of our main package so you should add both dependencies into your gradle file.

Reviewer 3: ```suggestion
**Note**: This package is an extension of the main package, so add both dependencies into your gradle file.
```

## Setup

```conf
repositories {
    maven { url "https://dl.bintray.com/datadog/datadog-maven" }
}

dependencies {
    implementation "com.datadoghq:dd-sdk-android:x.x.x"
    implementation "com.datadoghq:dd-sdk-android-ndk:x.x.x"
}
```

### Plugins

> 1. The NDKCrashReporterPlugin handles the crash signals at the native level and reports them as errors by adding more explicity information (e.g. backtrace, signal name, signal relevant error message) in the Datadog Logs dashboard:

Reviewer 3: ```suggestion
1. The NDKCrashReporterPlugin handles the crash signals at the native level and reports them as errors by adding more explicit information (for example: backtrace, signal name, signal relevant error message) in the Datadog logs dashboard:
```
